hedera-node/src/main/java/com/hedera/services/state/submerkle/CustomFeesBalanceChange.java

```diff
@@ -0,0 +1,181 @@
+package com.hedera.services.state.submerkle;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hederahashgraph.api.proto.java.AccountAmount;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import proto.CustomFeesOuterClass;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Process self serializable object that encapsulates a balance change, either ℏ or token unit.
```

Comment:
Along with the rename to `AssessedCustomFee`, would emphasize that this type represents an **assessed** custom fee, which may or may not result in a balance change, depending on if the payer can afford it.

Reply:
Changed javadoc. Thanks @tinker-michaelj